account/lnurl.go

```diff
@@ -29,18 +31,47 @@ type LoginResponse struct {
 	Token string `json:"token"`
 }
 
-func (a *Service) HandleLNURL(rawString string) (*data.LNUrlResponse, error) {
+func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err error) {
+
+	var iparams lnurl.LNURLParams
+	var rawurl, lightningAddress string
+
+	handleLNUrlError := fmt.Errorf("%q does not contain an LNURL.", rawString)
+
 	encodedLnurl, ok := lnurl.FindLNURLInText(rawString)
-	if !ok {
-		return nil, fmt.Errorf("'%s' does not contain an LNURL.", rawString)
-	}
+	if ok {
```

Comment:
When the rawString is an email address the only difference is an additional step to construct the lnurl URL.
Why not use this step only when the `FindLNURLInText` fails and then both flows merge to the same point at `lnurl.HandleLNURL`
We can also extract the url construction logic to its own function.

Something along the lines of:

```
if !ok {
  encodedURL, err := lnurlFromEmail(rawString)
  if err != nil {
    return nil, err
  }
}

rawurl, iparams, err := lnurl.HandleLNURL(encodedLnurl)
```